### PR TITLE
LoadingComponent created: Fix #299

### LINKED ISSUE #299: [Common] Create LoadingComponent
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the COMET-WEB
- [x] I have searched [open](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

A LoadingComponent should be used to warn the user that something is loading and aware him that the application is not frozen/not responsive. 

The LoadingComponent of [UI-DSM](https://github.com/RHEAGROUP/UI-DSM/tree/development/UI_DSM.Client/Components/App/LoadingComponent) could be used for inspiration
### Logs

  - [ ] There are no logs regarding the issue
  - [ ] I have attached a log file or included the relevant section in this issue  
  
### System Configuration

- COMET version:
  - [ ] COMET-WEB: 
  - [ ] Other: 
- Web Services:
  - [ ] COMET Web Services version:

### PR DESCRIPTION
**Pull request by antoineatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-WEB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-WEB-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

Fix #299

- LoadingComponent created. Needed to have async flow to have the LoadingComponent showing.
- All SingleIterationApplicationBaseViewModel and associated component can have a loading state

### CODE REVIEW

File: COMETwebapp/Components/Shared/LoadingComponent.razor.css [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+.splash-screen-caption {
+    font-size: 1.5rem;
+    font-weight: 600;
+    margin-top: 1.5rem;
+}
+
+.splash-screen-text {
+    color: white;
+    margin-top: .5rem;
+}
```
Comment (jaimeatrhea) [inline COMETwebapp/Components/Shared/LoadingComponent.razor.css:36]: New line at the end

File: COMETwebapp/ViewModels/Components/ParameterEditor/ParameterEditorBodyViewModel.cs [hunk 2]
```diff
@@ -117,39 +117,45 @@ public ParameterEditorBodyViewModel(ISessionService sessionService, ISubscriptio
...
 
         /// <summary>
         /// Initializes the <see cref="ParameterEditorBodyViewModel"/>
         /// </summary>
-        public void InitializeViewModel()
+        /// <returns>A <see cref="Task"/></returns>
+        public async Task InitializeViewModel()
         {
+            this.IsLoading = true;
+            await Task.Delay(1);
```
Comment (jaimeatrhea) [inline COMETwebapp/ViewModels/Components/ParameterEditor/ParameterEditorBodyViewModel.cs:153]: Why is a Delay?

File: COMETwebapp/ViewModels/Components/Viewer/ViewerBodyViewModel.cs [hunk 3]
```diff
@@ -93,48 +93,45 @@ public ViewerBodyViewModel(ISessionService sessionService, ISelectionMediator se
...
         /// <summary>
         /// Initializes this <see cref="IViewerBodyViewModel"/>
         /// </summary>
-        public void InitializeViewModel()
+        /// <returns>A <see cref="Task"/></returns>
+        public async Task InitializeViewModel()
         {
-            this.Elements = this.InitializeElements().ToList();
+            this.IsLoading = true;
+            await Task.Delay(1);
```
Comment (jaimeatrhea) [inline COMETwebapp/ViewModels/Components/Viewer/ViewerBodyViewModel.cs:130]: Same question as previous question

**Review by samatrhea — APPROVED** [review]

File: COMETwebapp/Components/Viewer/ViewerBody.razor.cs [hunk 4]
```diff
@@ -68,19 +66,23 @@ protected override async Task OnAfterRenderAsync(bool firstRender)
 
             if (firstRender)
             {
-                await base.OnInitializedAsync();
-                this.ViewModel.InitializeViewModel();
```
Comment (jaimeatrhea) [inline COMETwebapp/Components/Viewer/ViewerBody.razor.cs:72]: Why remove this line of code?

Comment (antoineatrhea) [reply COMETwebapp/Components/Viewer/ViewerBody.razor.cs:72]: The VM is initalized when setting the Iteration

**Review by jaimeatrhea — APPROVED** [review]

**Pull request merged**